sqld/src/query.rs

```diff
@@ -120,6 +120,12 @@ pub struct Row {
     pub values: Vec<Value>,
 }
 
+impl Row {
+    pub fn size_hint(&self) -> usize {
+        self.values.iter().map(|v| v.size_hint()).sum()
```

Comment:
When we are at it, we can also estimate the overhead of row encoding, which we can simplify to be 1 additional byte per column (for the comma that separates values in an array) and 3 additional bytes per row (for the `[]` that delimit the array of columns and for the comma that separates rows).

This overhead might be meaningful if the query returns a huge number of rows.